dartsim/src/SimulationFeatures.cc

```diff
@@ -52,7 +59,54 @@ void SimulationFeatures::WorldForwardStep(
 
   // TODO(MXG): Parse input
   world->step();
-  // TODO(MXG): Fill in output and state
+  this->WriteRequiredData(_h);
+  this->Write(_h.Get<JointPositions>());
+  // TODO(MXG): Fill in state
+}
+
+void SimulationFeatures::Write(JointPositions &/*_positions*/) const
+{
+  // TODO(adlarkin) implement this, if it's needed?
+}
+
+void SimulationFeatures::Write(WorldPoses &_poses) const
+{
+  // cache link poses from the previous iteration
+  // (using a static variable for now instead of a member variable
+  // since this method has to be const)
+  static std::unordered_map<std::size_t, math::Pose3d> prevLinkPoses;
+
+  // remove link poses from the previous iteration
+  _poses.entries.clear();
+
+  for (const auto &link : this->links.idToObject)
+  {
+    const auto id = link.first;
+    const auto info = link.second;
+
+    // make sure the link exists
+    if (info && info->link)
+    {
+      WorldPose wp;
+      wp.pose = ignition::math::eigen3::convert(
+          info->link->getWorldTransform());
+      wp.pose.Pos() = ignition::math::eigen3::convert(
+          info->link->getCOM());
```

Comment:
This was inspired by the [DARTDoublePendulum test](https://github.com/ignitionrobotics/ign-physics/blob/c77ac49b01e7a57cd0a6f6ba76c46837c65210ab/test/plugins/DARTDoublePendulum.cc#L307-L309). Is there a reason why we need to set the pose's position to the link's center of mass, instead of just using the result from `getWorldTransform` in the previous line?